src/connections/SitePolygons.ts

```diff
@@ -0,0 +1,76 @@
+import { useMemo } from "react";
+import { createSelector } from "reselect";
+
+import {
+  sitePolygonsIndex,
+  SitePolygonsIndexQueryParams
+} from "@/generated/v3/researchService/researchServiceComponents";
+import {
+  sitePolygonsIndexFetchFailed,
+  sitePolygonsIndexIsFetching
+} from "@/generated/v3/researchService/researchServicePredicates";
+import { SitePolygonDto } from "@/generated/v3/researchService/researchServiceSchemas";
+import { ApiDataStore } from "@/store/apiSlice";
+import { Connection } from "@/types/connection";
+import { connectionHook, connectionLoader } from "@/utils/connectionShortcuts";
+
+export type SitePolygonConnectionParams = {
+  entityName: string;
+  entityUuid: string;
+};
+
+export type SitePolygonConnection = {
+  sitePolygons?: SitePolygonDto[] | undefined;
+  sitePolygonsHaveFailed: boolean;
+  sitePolygonsAreLoading: boolean;
+};
+
+const sitePolygonQuery = (entityName: string, entityUuid: string) => {
+  const queryParams = {
+    "page[number]": 0, // update once pagination is implemented
+    "page[size]": 10
+  } as SitePolygonsIndexQueryParams;
+  if (entityName === "projects") {
+    queryParams["projectId[]"] = [entityUuid];
+  } else if (entityName === "sites") {
+    queryParams["siteId[]"] = [entityUuid];
+  }
+  return { queryParams };
+};
+const sitePolygonsSelector = (store: ApiDataStore) =>
+  Object.values(store.sitePolygons || {}).map(resource => resource.attributes);
+
+const sitePolygonsConnection = (params: SitePolygonConnectionParams): Connection<SitePolygonConnection> => ({
+  load: ({ sitePolygons }) => {
+    if (sitePolygons == null || sitePolygons.length === 0) {
+      const queryParams = sitePolygonQuery(params.entityName, params.entityUuid);
+      sitePolygonsIndex(queryParams);
+    }
+  },
+  isLoaded: ({ sitePolygons, sitePolygonsHaveFailed }) => sitePolygons !== null || sitePolygonsHaveFailed,
+  selector: createSelector(
+    [
+      sitePolygonsSelector,
+      sitePolygonsIndexIsFetching(sitePolygonQuery(params.entityName, params.entityUuid)),
+      sitePolygonsIndexFetchFailed(sitePolygonQuery(params.entityName, params.entityUuid))
+    ],
+    (sitePolygons, sitePolygonsAreLoading, sitePolygonsFailed) => ({
+      sitePolygons,
```

Comment:
This is a little too simple. According to `sitePolygonsSelector` current implementation, this will return all of the site polygons in the store every time. This connection should be filtering out the ones that don't apply to the site / project current being requested for this use of the connection.

Reply:
Yeah, mapping off the index meta ids does this for you.